cpp/src/arrow/dataset/scanner.h

```diff
@@ -240,6 +240,9 @@ class ARROW_DS_EXPORT ScannerBuilder {
   /// \brief Set the pool from which materialized and scanned arrays will be allocated.
   Status Pool(MemoryPool* pool);
 
+  /// \brief Set fragment-specific scan options.
+  Status FragmentScanOptions(std::shared_ptr<FragmentScanOptions> fragment_scan_options);
```

Comment:
Just ScanOptions instead of FragmentScanOptions might be more descriptive? Because it's not that this can be set for each fragment. It's the same for all fragments *for one scan*.

Reply:
That's fair, but that does collide with ScanOptions itself, unless you mean just the naming of the builder method?

Reply:
Ah, I see…hmm, but if we had a hypothetical FlightFragment, we'd still want to have scan options specific to that fragment, right?

Reply:
That would be "FlightScanOptios" then?

Reply:
Yeah, I think the slight niggle I have there is that there wouldn't be a corresponding 'Flight(File)Format'. Maybe 'PerScanOptions'? But it's not a big deal and FormatScanOptions is OK with me too.

Reply:
Ah, OK, I see now that the "Fragment" in the name was meant to mean the general "fragment *type*", while I interpreted it as the "single fragment". 
Anyway, this is more a nitpicky remark, not too important ;)